### PR TITLE
add resourceApply for prometheus rule

### PR DESCRIPTION
**Pull request by deads2k**
To handle future us adding alerts to our operator.

/hold pending proof

### CODE REVIEW

**Comment by deads2k** [comment]
proof in https://github.com/openshift/cluster-kube-apiserver-operator/pull/1073

File: pkg/operator/resource/resourceapply/monitoring.go [hunk 1]
```diff
@@ -86,3 +86,48 @@ func ApplyServiceMonitor(client dynamic.Interface, recorder events.Recorder, req
...
+	}
+
+	if klog.V(4).Enabled() {
+		klog.Infof("PrometheusRule %q changes: %v", namespace+"/"+required.GetName(), JSONPatchNoError(existing, existingCopy))
+	}
+
+	newObj, err := client.Resource(prometheusRuleGVR).Namespace(namespace).Update(context.TODO(), updated, metav1.UpdateOptions{})
+	if err != nil {
+		recorder.Warningf("PrometheusRuleUpdateFailed", "Failed to update PrometheusRule.monitoring.coreos.com/v1: %v", err)
+		return nil, true, err
```
Comment (tkashem) [inline pkg/operator/resource/resourceapply/monitoring.go:128]: does the `bool` indicate `was updated or created`? I know the caller should not look at the `bool` or  the object if there is an error, but was wondering if it is semantically more correct to return `nil, false, err` here?

**Pull request merged**